docs/src/main/sphinx/functions/json.rst

```diff
@@ -2,6 +2,1174 @@
+  in JSON path expressions are zero-based.
+
+.. code-block:: text
+
+    last
+
+arithmetic binary expressions
+'''''''''''''''''''''''''''''
+
+The JSON path language supports five arithmetic binary operators:
+
+.. code-block:: text
+
+    <path1> + <path2>
+    <path1> - <path2>
+    <path1> * <path2>
+    <path1> / <path2>
+    <path1> % <path2>
+
+Both operands, ``<path1>`` and ``<path2>``, are evaluated to sequences of
+items. For arithmetic binary operators, it is required that both sequences
+consist of singleton numeric items. The arithmetic operation is performed
+accordingly to the SQL semantics, returning a singleton sequence containing
+the result.
+
+The operators follow the usual precedence, and parentheses can be used for
+grouping.
+
+arithmetic unary expressions
+''''''''''''''''''''''''''''
+
+.. code-block:: text
+
+    + <path>
+    - <path>
+
+The operand ``<path>`` is evaluated to a sequence of items. It is required that
+every item is a numeric value. The unary plus or minus is applied to every item
+in the sequence, following SQL semantics, and the results form the returned
+sequence.
+
+The remaining JSON path operations are collectively known as **accessors**.
+
+member accessor
+'''''''''''''''
+
+The member accessor returns the value of the member with the specified key for
+each JSON object in the input sequence.
+
+.. code-block:: text
+
+    <path>.key
+    <path>."key"
+
+The condition when a JSON object does not have such a member is called a
+structural error. In the lax mode, it is suppressed, and the faulty object is
+excluded from the result.
+
+Let ``<path>`` return a sequence of three JSON objects:
+
+.. code-block:: text
+
+    {"customer" : 100, "region" : "AFRICA"},
+    {"region" : "ASIA"},
+    {"customer" : 300, "region" : "AFRICA", "comment" : null}
+
+the expression ``<path>.customer`` succeeds in the first and the third object,
+but the second object lacks the required member. In strict mode, path
+evaluation fails. In lax mode, the second object is silently skipped, and the
+resulting sequence is ``100, 300``.
+
+It is required that all items in the input sequence are JSON objects.
+
+.. note::
+
+    Trino implementation does not support JSON objects with duplicate keys.
+
+wildcard member accessor
+''''''''''''''''''''''''
+
+Returns all bound values for each JSON object in the input sequence. All the
+partial results are concatenated into the returned sequence.
+
+.. code-block:: text
+
+    <path>.*
+
+Let ``<path>`` return a sequence of three JSON objects:
+
+.. code-block:: text
+
+    {"customer" : 100, "region" : "AFRICA"},
+    {"region" : "ASIA"},
+    {"customer" : 300, "region" : "AFRICA", "comment" : null}
+
+The results is:
+
+.. code-block:: text
+
+    100, "AFRICA", "ASIA", 300, "AFRICA", null
+
+It is required that all items in the input sequence are JSON objects.
+
+The order of the returned values follows the order of the original JSON
```

Comment:
This is not very clear. "original JSON objects" refers to the objects matched by the path. "within objects" refers to the wildcard matches within each of those objects. Maybe elaborate a little further to clarify that in the text.

Reply:
I tried to reword. I might need some more help with this piece.